### PR TITLE
Codechange: move all date-related variables inside the timer

### PR DESCRIPTION
**Pull request by TrueBrain**
## Motivation / Problem

While reading #10700, it became apparent that we have a few date-related globals that are now part of `TimerGameCalendar`, but are stored in `date.cpp`. To make code more readable and understandable, move them inside the class.

## Description

Simple search/replace for most part. Took some extra effort to make sure `date_func.h` is only included when actually needed.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by PeterN** [comment]
You already mentioned this in Discord, but also tick counter should not be in here :)

**Comment by TrueBrain** [comment]
> You already mentioned this in Discord, but also tick counter should not be in here :)

Indeed! Will fix that in a follow-up PR, if nobody minds. Makes it a bit more clear what moves to where :)

**Comment by PeterN** [comment]
Hmm, I thought a separate PR and remove the change from this one would make sense?

**Comment by TrueBrain** [comment]
> Hmm, I thought a separate PR and remove the change from this one would make sense?

SO MUCH WORK! /me grumbles a bit ... FINE. There. In a separate commit so you can witness the horror :P

File: src/tree_cmd.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "core/random_func.hpp"
 #include "newgrf_generic.h"
 #include "date_func.h"
+#include "timer/timer_game_calendar.h"
```
Comment (2TallTyler) [inline src/tree_cmd.cpp]: Leftover #include

File: src/newgrf_animation_base.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "animated_tile_func.h"
 #include "core/random_func.hpp"
 #include "date_func.h"
+#include "timer/timer_game_calendar.h"
```
Comment (2TallTyler) [inline src/newgrf_animation_base.h]: Leftover #include

File: src/gamelog.cpp [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #include "console_func.h"
 #include "debug.h"
 #include "date_func.h"
+#include "timer/timer_game_calendar.h"
```
Comment (2TallTyler) [inline src/gamelog.cpp]: Leftover #include

**Review by 2TallTyler — COMMENTED** [review]
Found some leftover `#include`s, otherwise looks good.

**Review by 2TallTyler — APPROVED** [review]

**Pull request merged**